app/preferencesdialog.cpp

```diff
@@ -581,6 +595,11 @@ void FilesPage::autosaveNumberChange(int number)
     mManager->set(SETTING::AUTO_SAVE_NUMBER, number);
 }
 
+void FilesPage::clearRecentFilesList(bool b)
+{
+    mManager->set(SETTING::CLEAR_RECENT_FILES_LIST, b);
+}
+
```

Comment:
I, too, am not 100% happy with using a preference key for this. I don’t think we should squeeze the button into this structure no matter what. Also, I got the success message two times once; I couldn’t reproduce it since then but I suspect it could be somehow related to this.